ibis/expr/types/relations.py

```diff
@@ -2966,13 +2966,16 @@ def join(
         # must be wrapped in SelfReferences so that we can join the same table
         # with itself multiple times and to enable optimization passes later on
         left = left.op()
-        if isinstance(left, ops.SelfReference):
-            left = left.parent
+        if isinstance(left, ops.JoinChain):
+            # if the left side is already a join chain, we can reuse it
+            expr = left.to_expr()
```

Comment:
How does this occur coming from a user-constructed expression? How can the left side be a join chain and the next call ends up here?